### PR TITLE
Fix Gemini tool calling via GeminiProvider

### LINKED ISSUE #78: Tool calling not working with Gemini model
In an Agent that does tool calling if using Gemini 2.5 pro after a long time the call to the model goes in 500 error. The same code works with: xAI grok4, OPenAI GPT-5, OpenAI GPT OSS 120B (all on OCI GenAI).

I have developed a simple test case that can be easily run:

[test case](https://github.com/luigisaetta/mcp-oci-integration/blob/main/tests/test_gemini_bug.py
)

To run it simply put your Compartment ocid in the python file.

With MODEL1 (gemini pro) you see the problem. Switching to MODEL2 you get quickly a correct answer from LLM

### PR DESCRIPTION
**Pull request by fede-kamel**
## Summary
Fixes #78 - Tool calling with Google Gemini models on OCI GenAI fails with 500 error.

cc @luigisaetta

## Problem
Tool calling with Google Gemini models (google.gemini-2.5-flash, google.gemini-2.5-pro) fails with HTTP 500 Internal Server Error after ~93 second timeout. The same OCI SDK request format works correctly for other models (xAI Grok, Meta Llama).

### 1. Same Request Format - Different Results
The identical OCI Generic API message format:
```python
messages = [
    UserMessage(content=[TextContent(text="What's the weather?")]),
    AssistantMessage(
        content=[TextContent(text="")],
        tool_calls=[FunctionCall(id="c1", name="get_weather", arguments='{"city":"Rome"}')]
    ),
    ToolMessage(
        content=[TextContent(text='{"temp":20}')],
        tool_call_id="c1"
    )
]
```

**Result with xAI Grok:** ✅ Works correctly, returns response in ~2s
**Result with Google Gemini:** ❌ 500 Internal Server Error after ~93s

### 2. Root Cause
The OCI GenAI service translation layer doesn't properly convert the Generic API format to Gemini's native format:
- **OCI Generic API:** Uses `role: "TOOL"` with `ToolMessage` 
- **Gemini Native API:** Expects `role: "user"` with `functionResponse` in `parts`

The OCI backend doesn't support `ToolMessage` or `tool_calls` in `AssistantMessage` for Gemini models.

### 3. Simple Chat Works
Basic chat without tool calling works fine with Gemini through OCI, confirming the issue is specific to tool message handling.

## Solution
Added `GeminiProvider` class that converts tool-related messages to regular user/assistant messages that Gemini can process:

- `AIMessage` with `tool_calls` → `AssistantMessage` with descriptive text  
- `ToolMessage` → `UserMessage` with tool result text

This workaround allows Gemini tool calling to function correctly through OCI's GenAI service until the backend properly supports the Generic API tool format.

### Parallel Tool Calling Tests (8 tests)
```
test_parallel_tool_results_same_function[gemini-2.5-flash] PASSED
test_parallel_tool_results_same_function[gemini-2.5-pro] PASSED
test_parallel_tool_results_different_functions[gemini-2.5-flash] PASSED
test_parallel_tool_results_different_functions[gemini-2.5-pro] PASSED
test_parallel_five_tool_calls[gemini-2.5-flash] PASSED
test_parallel_five_tool_calls[gemini-2.5-pro] PASSED
test_parallel_tool_calls_with_errors[gemini-2.5-flash] PASSED
test_parallel_tool_calls_with_errors[gemini-2.5-pro] PASSED
```

### Additional Coverage
- Cross-model compatibility tests (Gemini, Meta, xAI)
- Single and multi-tool result processing
- Tool binding and invocation
- Edge cases (empty results, error handling)

## Files Changed
- `langchain_oci/chat_models/oci_generative_ai.py` - Added GeminiProvider class
- `tests/integration_tests/chat_models/test_gemini_tool_calling.py` - Comprehensive integration tests (800+ lines)

## Backward Compatibility
✅ **No Breaking Changes** - Existing functionality unchanged, only adds new provider for Gemini models

### CODE REVIEW

**Comment by fede-kamel** [comment]
Closing this PR in favor of the simpler approach in PR #79.

## Summary

This PR successfully demonstrated that Gemini tool calling can work through OCI GenAI with proper message handling. The comprehensive test suite (38 integration tests including real agent loops) proved the concept works well.

However, after testing PR #79's approach, I found that the 13-line fix addresses the root causes:
- UUID generation for `id: None` cases
- Empty content handling

Both approaches work, but PR #79 achieves the same result with significantly less complexity.

## Why the simpler approach

- **Maintainability**: Single code path for all models vs. separate Gemini provider
- **Future-proofing**: Preserves structured tool calls for new LangChain features
- **Less technical debt**: Works with the platform rather than around it

## Outcome

The Gemini tool calling issue is resolved via PR #79. Thanks to @paxiaatucsdedu for the quick fix and @luigisaetta for highlighting this issue!

The extensive testing and agent loop validation in this PR was valuable for confirming the fix works end-to-end.

File: libs/oci/langchain_oci/chat_models/oci_generative_ai.py [hunk 1]
```diff
@@ -1105,6 +1105,169 @@ class MetaProvider(GenericProvider):
     pass
 
 
+class GeminiProvider(GenericProvider):
```
Comment (sshravan91) [inline libs/oci/langchain_oci/chat_models/oci_generative_ai.py]: This needs to implement `chat_stream_to_text` to support streaming.

`ChatOCIGenAI` exposes `_stream` that internally invokes, `self._provider.chat_stream_to_text(event_data)`
